radicle-surf/src/vcs/git/repo.rs

```diff
@@ -441,97 +441,99 @@ impl<'a> RepositoryRef<'a> {
         Ok(diff)
     }
 
-    fn update_file_map(
-        path: file_system::Path,
-        name: file_system::Label,
-        file: directory::File,
-        files: &mut HashMap<file_system::Path, NonEmpty<(file_system::Label, directory::File)>>,
-    ) {
-        files
-            .entry(path)
-            .and_modify(|entries| entries.push((name.clone(), file.clone())))
-            .or_insert_with(|| NonEmpty::new((name, file)));
-    }
+    /// Generates a Directory for the commit.
+    fn directory_of_commit(&self, commit: &git2::Commit) -> Result<Directory, Error> {
+        let mut parent_dirs = vec![Directory::root()];
+        let git2_tree = commit.as_object().peel_to_tree()?;
```

Comment:
nit: we don't need the extra `git2` naming :) Editors can show the type if someone needs to know where it comes from
```suggestion
        let tree = commit.as_object().peel_to_tree()?;
```